src/automations/checkInactivity.js

```diff
@@ -30,6 +30,7 @@ async function scrapePullRequests(client, pullRequests) {
     const number = pullRequest.number;
     const repoName = pullRequest.base.repo.name;
     const repoOwner = pullRequest.base.repo.owner.login;
+    const reviewedLabel = pullRequest.reviewed.label;
```

Comment:
I just noticed this wasn't what I originally thought this was. `pullRequest` requests a single object in the array returned from `client.pullRequests.getAll()` (see [here](https://developer.github.com/v3/pulls/#list-pull-requests) for payload example), and there's no reviewed property of a pull request.

Reply:
In `events/pullRequests.js` 
```
const response = await client.issues.getIssueLabels({
    owner: repoOwner, repo: repoName, number: number
  });

  let labels = response.data.map(label => label.name);
  const needsReviewLabel = client.cfg.activity.pullRequests.needsReview.label;
  const reviewedLabel = client.cfg.activity.pullRequests.reviewed.label;
  const needsReview = labels.includes(needsReviewLabel);
  const reviewed = labels.includes(reviewedLabel);
```

This piece of code gets the issue, its labels and then accesses `client.cfg.activity.pullRequests.needsReview.label` to look for the review label.

Something similar could be done for `checkInactivity` since `issues` is being passed to `scrapeInactiveIssues`. After checking whether the reference to the issues is present, we could check for the presence of `const needsReviewLabel = client.cfg.activity.pullRequests.needsReview.label`. Thoughts?

Reply:
Actually, in the `checkInactivePullRequest` we do individual PR label checks for the pull requests. I think it might be best to have the `checkInactivePullRequest` function return a Boolean value representing whether the function found the reviewedLabel there and assign to the data object accordingly.

Reply:
Yes, that was what i was thinking of :D 
So, something like `if (!inactiveLabel && !reviewedLabel) return true;` could be done in `checkInactivePullRequest`. 

But it is being called only when there has been no activity on the PR beyond the reminder clock. To pass the bool value up through `checkInactivePullRequest` it needs to be called outside the time check as well

For instance 
```
if (checkInactivePullRequest(client, pullRequest)) {
      reviewedLabel = checkInactivePullRequest(client, pullRequest);
    }
``` 
here `true` is returned in the if block and is set to `reviewedLabel` which is then passed into `data` object. But `checkInactivePullRequest` is being called twice here which might not be desirable

Reply:
Maybe it'd be best to pull the label checking functionality out from the `checkInactivePullRequest` function then?

Reply:
We could create a separate function which carries out the task of extracting the inactive and reviewed labels from the pull request. Inside that very function, we could return an object with those two labels, and access them elsewhere. This way, the function can be called both inside `checkInactivePullRequest` and outside it, wherever else needed; in this case in `scrapePullRequests`
```
const labels = await client.issues.getIssueLabels({
    owner: repoOwner, repo: repoName, number: number
  });
  const inactiveLabel = labels.data.find(l => {
    return l.name === client.cfg.activity.inactive;
  });
  const reviewedLabel = labels.data.find(l => {
    return l.name === client.cfg.activity.pullRequests.reviewed.label;
  });
```
This piece of code could be put into the function which would return an object containing the labels. This function can then be called inside as well as outside `checkInactivePullRequest`

Reply:
Sure, that sounds reasonable to implement :) 👍